core/spec/lib/spree/event/adapters/default_spec.rb

```diff
@@ -99,6 +99,50 @@ def toggle
 
             expect(dummy.box).to eq('foo')
           end
+
+          it 'adds the fired event with given caller location to the firing result object' do
+            bus = described_class.new
+            dummy = Class.new do
+              attr_reader :run
+
+              def initialize
+                @run = false
+              end
+
+              def toggle
+                @run = true
+              end
+            end.new
+            bus.subscribe('foo') { dummy.toggle }
+
+            firing = bus.fire 'foo', caller_location: caller_locations(0)[0]
+
+            expect(firing.event.caller_location.to_s).to include(__FILE__)
+          end
+
+          it 'adds the triggered executions to the firing result object' do
+            bus = described_class.new
+            dummy = Class.new do
```

Comment:
Also here, I think we can avoid introducing a new class definition and reuse the one(s) from above, because it seems to me that this spec doesn't depend on the logic of this class.